packages/web/app/src/components/project/alerts/create-channel.tsx

```diff
@@ -42,186 +100,258 @@ export const CreateChannelModal = ({
-              <div className="text-sm text-red-500">{errors.endpoint}</div>
-            )}
-            {mutation.data?.addAlertChannel.error?.inputErrors.webhookEndpoint && (
-              <div className="text-sm text-red-500">
-                {mutation.data.addAlertChannel.error.inputErrors.webhookEndpoint}
-              </div>
-            )}
-            {values.endpoint ? (
-              <p className="text-sm text-gray-500">Hive will send alerts to your endpoint.</p>
-            ) : (
-              <a href="https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=newteams%2Cdotnet">
-                Follow this guide to set up an incoming webhook connector in MS Teams
-              </a>
-            )}
-          </div>
-        )}
+    if (values.endpoint && !values.endpoint.startsWith('http')) {
+      form.setError('endpoint', { message: 'Endpoint should be a valid URL' });
+      return;
+    }
 
-        {values.type === AlertChannelType.Slack && (
-          <div className="flex flex-col gap-4">
-            <label className="text-sm font-semibold" htmlFor="endpoint">
-              Slack Channel
-            </label>
-            <Input
-              name="slackChannel"
-              value={values.slackChannel}
-              onChange={handleChange}
-              onBlur={handleBlur}
-              placeholder="Where should Hive post messages?"
-              disabled={isSubmitting}
-              isInvalid={touched.slackChannel && !!errors.slackChannel}
-              className="grow"
-            />
-            {touched.slackChannel && errors.slackChannel && (
-              <div className="text-sm text-red-500">{errors.slackChannel}</div>
-            )}
-            {mutation.data?.addAlertChannel.error?.inputErrors.slackChannel && (
-              <div className="text-sm text-red-500">
-                {mutation.data.addAlertChannel.error.inputErrors.slackChannel}
-              </div>
-            )}
-            <p className="text-sm text-gray-500">
-              Use <Tag>#channel</Tag> or <Tag>@username</Tag> form.
-            </p>
-          </div>
-        )}
+    const { data, error } = await mutate({
+      input: {
+        name: values.channelName,
+        type: values.type,
+        organization: organizationId,
+        project: projectId,
+        webhook: isWebhookLike && values.endpoint ? { endpoint: values.endpoint } : null,
+        slack:
+          values.type === AlertChannelType.Slack && values.slackChannel
+            ? { channel: values.slackChannel }
+            : null,
+      },
+    });
+    // When Data is null, it means that the user has not integrated with Slack App
+    if (!data) {
+      toggleModalOpen();
+      await toast({
+        title: 'Please integrate with Slack App',
+        description:
+          'The channel has been created but you will not receive alerts until you integrate with Slack App',
```

Comment:
Hmm, what does it actually means? should the user do something? should we point them somewhere? 🤔 

Also, I think if you just check `!data` first, all Mutation errors will fall into that condition, so the other `else if` will never get called?